src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +115,123 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
+                        BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                        entries.add(bibEntry);
                     }
                 } else {
                     var element = node.getObject();
-                    JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
-                    for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bibTexFormat");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+
+                    if (element.has("all")) {
+                        JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
+                    } else if (element.has("results")) {
+                        JSONArray entriesArray = element.getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
                     }
                 }
-            } catch (JSONException | TokenMgrException e) {
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+            // Set fields based on the mappings
+            for (Map.Entry<StandardField, List<String>> mapEntry : fieldMappings.entrySet()) {
+                StandardField field = mapEntry.getKey();
+                List<String> path = mapEntry.getValue();
+
+                String value;
+                if (field == StandardField.AUTHOR) {
+                    value = toAuthors(item.optJSONArray(path.getFirst()));
+                } else if (field == StandardField.KEYWORDS) {
+                    value = getKeywords(item.optJSONObject(path.getFirst()));
+                } else {
+                    value = getOrNull(item, path).orElse(null);
+                }
+
+                if (value != null) {
+                    entry.setField(field, value);
+                }
+            }
+            // Handle articleUrl and mrnumber fields separately
```

Comment:
Can you say "why" here?

Reply:
So if we take a look at the general json structure, for example:
![image](https://github.com/JabRef/jabref/assets/74734844/37b02c9f-9060-469d-88eb-0167713a00dd)
We can see that mrnumber and articleurl are direct key:value mappings, so the strings could be obtained directly. Whereas for the rest of the fields, such as title or authors, we had to navigate their respective nested structures.
We could've handled them uniformly as well, should I change it?

Reply:
Just add a Java comment (maybe a bit shorter than your text)